### PR TITLE
New auto re-auth functionality added

### PR DESCRIPTION
**Pull request by lawinslow**

### CODE REVIEW

**Comment by lawinslow** [comment]
@aappling-usgs a subsequent mda.streams PR is coming

File: R/session_check_reauth.R [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#'@title Checks current session and re-authenticates if necessary
+#'
+#'@inheritParams authenticate_sb
+#'@param session (Advanced) Alternatively supply a specific session to be checked
+#'@param ... Any additional parameters are ignored. 
+#'
+#'@description
+#'This checks the state of your Sciencebase session and, if invalid or if you've supplied
+#'a new user name, it will re-authenticate. This is primarily a helper function for 
+#'packages that reverse-depend on sbtools.
```
Comment (aappling-usgs) [inline R/session_check_reauth.R]: reverse-depend?

File: R/session_check_reauth.R [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#'session_check_reauth('user@@usgs.gov')
+#'
+#'#default behavior
+#'session_check_reauth(session=current_session())
+#'
+#'}
+#'
+#'
+#'@export
+session_check_reauth = function(username, password, session, ...){
```
Comment (aappling-usgs) [inline R/session_check_reauth.R]: so use this in place of authenticate_sb, right? call it just before running any sbtools commands that require authentication? or do sbtools functions that need authentication already call this themselves?

**Comment by aappling-usgs** [comment]
@lawinslow awesome on the mda.streams PR - i'll sit tight.

File: R/session_check_reauth.R (hunk 2 above)
Comment (lawinslow) [inline R/session_check_reauth.R]: Hmm, could have sbtools call it. Need to break some interfaces in sbtools. 

The question is, is it better to have sbtools pushing user dialog interactions to the top, or to have it operate more fundamentally and later tools handle such user interaction?

File: R/session_check_reauth.R (hunk 1 above)
Comment (lawinslow) [inline R/session_check_reauth.R]: No, that's wrong, don't know what I was thinking

File: R/session_check_reauth.R (hunk 2 above)
Comment (lawinslow) [inline R/session_check_reauth.R]: I'm going to leave it out of sbtools. I want an user to specifically choose this functionality as, usually, I don't assume most of my function calls are going to try and create a user dialog. Putting this into the sbtools functions would embed interactive logic very deeply into the package, right now interactive code sits near the surface, in two functions, authenticate_sb and now this new function.

Comment (aappling-usgs) [inline R/session_check_reauth.R]: agreed

**Pull request merged**